lib/spack/env/cc

```diff
@@ -142,16 +142,20 @@ esac
 # If any of the arguments below are present, then the mode is vcheck.
 # In vcheck mode, nothing is added in terms of extra search paths or
 # libraries.
```

Comment:
This looks great, but I think it would be good to update this comment like

> Look for option to print version (in which case the mode is vcheck). For ld this is -v. For most other compiler binaries it is -V, --version, or -dumpversion. 

> (same as before) In vcheck mode, nothing is added in terms of extra search paths or libraries.

> TODO: For the xl compiler, -V indicates verbose output, so we should not enable vcheck mode for -V when using xl

A couple ways that might be easy to check:

* Examine `SPACK_CC`
* Pass in a new variable like `SPACK_VERBOSE_FLAG`, use `compiler.verbose_flag` to define it, and don't enable vcheck mode if `$arg` is `$SPACK_VERBOSE_FLAG`

That doesn't have to happen in this PR though (although in that case modifying the comment as suggested above is important).